frontend/src/app/[locale]/subscribe/actions.ts

```diff
@@ -0,0 +1,137 @@
+"use server";
+
+import { environment } from "src/constants/environments";
+import { z } from "zod";
+
+import { getTranslations } from "next-intl/server";
+import { redirect } from "next/navigation";
+
+import { ValidationErrors } from "src/components/subscribe/SubscriptionForm";
+
+type sendyResponse = {
+  validationErrors: ValidationErrors;
+  errorMessage: string;
+};
+
+export async function subscribeEmail(
+  prevState: sendyResponse,
+  formData: FormData,
+) {
+  const t = await getTranslations("Subscribe");
+
+  const { errorMessage, validationErrors } = await subscribeEmailAction(
+    t,
+    formData,
+  );
+  if (errorMessage ?? validationErrors) {
+    return {
+      errorMessage,
+      validationErrors,
+    };
+  }
+  // Navigate to the sub confirmation page if no error returns short circuit the function
+  redirect(`/subscribe/confirmation`);
+}
+
+export async function subscribeEmailAction(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  t: any,
+  formData: FormData,
+): Promise<sendyResponse> {
+  const schema = z.object({
+    name: z.string().min(1, {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
```

Comment:
Looks like it is [discouraged by the maintainer](https://github.com/amannn/next-intl/discussions/1134#discussioncomment-9796463) but there is a [better way](https://github.com/amannn/next-intl/discussions/1134#discussioncomment-10940240). Let's try that.

Reply:
Very weird. It seems like they're saying they discourage passing the function around _because_ the typing is cumbersome, rather than that they specifically left the typing cumbersome in order to discourage passing the function around for some more thought out reason.

I can see not wanting to pass this function around for one particular reason: if you pass `t` from a server component to a client component you end up potentially closing over the entire messages object, and shipping the entire text content of your site to the client unnecessarily. I'm not exactly sure how next-intl works or if that is expected as the cost of doing business but it seems potentially, and this is a very scientific word here, slops.